From 4c6ce66f252ef0e063734838a41c4ec74f922581 Mon Sep 17 00:00:00 2001
From: ROCKNIX <hello@rocknix.org>
Date: Tue, 15 Sep 2026 19:10:00 +0800
Subject: [PATCH] usb: only set shutdown on supported kernels

The Rockchip 5.10 BSP's struct usb_driver has no shutdown callback.
The existing version range incorrectly enables it for that kernel and
prevents the module from compiling.
---
 os/linux/hif/usb.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/os/linux/hif/usb.c b/os/linux/hif/usb.c
index a89f90d..54ce42f 100644
--- a/os/linux/hif/usb.c
+++ b/os/linux/hif/usb.c
@@ -1110,8 +1110,6 @@ static struct usb_driver zt_usb_driver =
     .reset_resume   =   NULL,//zt_usb_resume,
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 11, 0))
     .shutdown       =   zt_usb_shutdown,
-#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19)) && (LINUX_VERSION_CODE < KERNEL_VERSION(6, 8, 0))
-    .shutdown    = zt_usb_shutdown,
 #endif
     .supports_autosuspend       = 1,
 };
-- 
2.43.0

